src/mathutil.cpp

```diff
@@ -9,6 +9,9 @@
 #include <cstdlib>
 #include "mathutil.hpp"
 
+std::mt19937 game_rand;
+std::mt19937 ui_rand;
```

Comment:
I think `ui_rand` can actually be a private static member variable of `cDialog`, since it's not used anywhere else… unless you think it'll be needed in other places?